include/private/target/buffer.h

```diff
@@ -50,12 +50,96 @@ struct buffer_target {
+ * 
+ * @param target A pointer to the `buffer_target` structure to be destroyed.
+ *               Must not be `NULL`.
+ *
+ * @note Ensure that the `target` is no longer accessed after calling this
+ *       function, as its memory will be deallocated.
+ */
 void
 destroy_buffer_target( const struct buffer_target *target );
 
+/**
+ * @brief Creates and initializes a new buffer target structure.
+ *
+ * This function allocates memory for a `buffer_target` structure and initializes
+ * its fields. The buffer target is designed to manage a pre-allocated buffer
+ * with specified size, enabling read and write operations.
+ *
+ * **Thread Safety: MT-Unsafe**
+ * It initializes a mutex (buffer_mutex) to protect shared resources in the
+ * buffer_target structure. This ensures that multiple threads can safely use
+ * the structure once it is properly initialized.
+ * 
+ * **Async Signal Safety: AS-Unsafe*
+ * It uses dynamic memory allocation (e.g., malloc or similar), which is not signal-safe. 
+ * 
+ * **Async Cancel Safety: AC-Unsafe**
+ * The function is not safe to call from threads that may be asynchronously canceled
+ * because the memory allocation (or its failure) might not handle thread cancellation properly
+ * 
+ * @param buffer A pointer to the pre-allocated memory buffer that this target
+ *               will manage. This buffer must be valid for the lifetime of the
+ *               `buffer_target` structure.
+ * @param size The size of the buffer in bytes.
+ *
+ * @return A pointer to the newly allocated and initialized `buffer_target`
+ *         structure, or `NULL` if memory allocation fails.
+ */
 struct buffer_target *
 new_buffer_target( char *buffer, size_t size );
 
+/**
+ * @brief Writes a message to a buffer target, wrapping around if needed.
+ *
+ * This function writes a message to the buffer target, ensuring that the buffer
+ * does not overflow. If the message is too large to fit in the buffer, an error
+ * is raised. The function also ensures thread safety by locking a mutex during
+ * the write operation.
+ * 
+ * **Thread Safety: MT-Unsafe**
+ * It ensures thread safety by locking a mutex during the write operation. 
+ * This prevents concurrent access to the shared buffer by multiple threads.
+ * 
+ * **Async Signal Safety: AS-Unsafe*
+ * It uses a non-reentrant lock (mutex) to coordinate access to the buffer.
+ * Mutex operations are not safe in signal handler contexts.
+ * 
+ * **Async Cancel Safety: AC-Unsafe**
+ * The function is not safe to call from threads that may be asynchronously canceled. 
+ * If a thread is canceled while holding the mutex, the mutex might not be released,
+ * potentially leading to deadlocks or inconsistent state
+ * 
+ * @param target A pointer to the `buffer_target` where the message will be written.
+ *               Must not be `NULL`.
+ * @param msg A pointer to the message to be written. The message should not
+ *            include a newline character at the end.
```

Comment:
The message is in fact expected to contain a newline at the end (which is stripped off by this function).